addons/calendar/static/src/js/calendar_view.js

```diff
@@ -18,6 +18,7 @@ var AttendeeCalendarPopover = CalendarPopover.extend({
         var self = this;
         this._super.apply(this, arguments);
         var session = this.getSession();
+        this.eventPrivacy = this.event.record.privacy === 'private' ? _.contains(this.event.record.partner_ids, session.partner_id) : true;
```

Comment:
Would have 2 public methods:
- `isEventPrivate()` (basically `this.event.record.privacy === 'private'`)
- `isCurrentPartnerAttendee()` (basically `this.event.record.partner_ids.includes(session.partner_id)`)

replace `showStatusDropdown` with `isCurrentPartnerAttendee()`